std/machines/small_field/pointer_arith.asm

```diff
@@ -0,0 +1,38 @@
+/// Increments a pointer by one full 32-bit word.
+///
+/// The pointers are given in 2 16-bit limbs. This function return the polynomials constraining
+/// (int(post_high) * 0x10000 + int(post_low)) to (int(pre_high) * 0x10000 + int(pre_low) + 4).
+///
+/// The returned polynomials are not automatically added as constraints. This is so the caller
```

Comment:
I think it would still be better to return a `Constr[]` instead of a `expr[]`. If you return `expr[]` you still need the assumption that they will be set to zero. You can use `std::constraints::make_conditional` to add a latch condition if you like.

Reply:
Done.